### PR TITLE
core: assorted tests and fixes

### PR DESCRIPTION
**Pull request by tvanderstad**
includes fixes for:
* drawing validation
* listing metadata, children, recursive children in the presence of links

### CODE REVIEW

File: core/tests/account_service_tests.rs [hunk 1]
```diff
@@ -99,6 +99,17 @@ fn import_account_corrupted() {
...
+fn import_account_corrupted_base64() {
+    let core = test_core();
+
+    base64::decode("clearlyabadaccountstring").unwrap();
+    assert!(matches!(
+        core.import_account("clearlyabadaccountstring"),
+        Err(Error::UiError(ImportError::AccountStringCorrupted))
+    ));
+}
+
```
Comment (steverusso) [inline core/tests/account_service_tests.rs:112]: Unsure about the base64 decode line

File: core/tests/account_service_tests.rs [hunk 2]
```diff
@@ -138,4 +149,23 @@ fn import_account_public_key_mismatch() {
...
+    core.create_account(&random_name(), &url()).unwrap();
+    core.export_account().unwrap();
+}
+
+#[test]
+fn export_account_qr() {
+    let core = test_core();
+    core.create_account(&random_name(), &url()).unwrap();
+    core.export_account_qr().unwrap();
+}
```
Comment (steverusso) [inline core/tests/account_service_tests.rs:164]: Can exporting qr just be added as the last line to the `export_account` test above?

File: core/src/model/drawing.rs [hunk 3]
```diff
@@ -19,15 +19,34 @@ use serde_json::error::Category;
 use crate::model::errors::core_err_unexpected;
 use crate::{CoreError, CoreResult};
 
+pub fn validate(drawing: &Drawing) -> CoreResult<()> {
+    for stroke in &drawing.strokes {
+        if stroke.points_x.len() != stroke.points_y.len()
+            || stroke.points_y.len() != stroke.points_girth.len()
+        {
+            return Err(CoreError::Unexpected(String::from("unequal points and girth metrics")));
```
Comment (Parth) [inline core/src/model/drawing.rs:27]: why not `CoreError::DrawingInvalid`?

File: core/src/model/drawing.rs [hunk 4]
```diff
@@ -88,45 +107,27 @@ pub fn export_drawing(
...
-
         for point_index in 0..stroke.points_x.len() - 1 {
             let mut pb = PathBuilder::new();
             let x1 = stroke
                 .points_x
                 .get(point_index)
-                .ok_or_else(|| {
-                    CoreError::Unexpected(String::from("unable to get color from alias"))
-                })?
+                .ok_or_else(|| CoreError::Unexpected(String::from("unable to get stroke x1")))?
```
Comment (Parth) [inline core/src/model/drawing.rs:115]: @SmailBarkouch / @tvanderstad why not `CoreError::DrawingInvalid`?

File: core/tests/account_service_tests.rs [hunk 5]
```diff
@@ -99,6 +99,17 @@ fn import_account_corrupted() {
     ));
 }
 
+#[test]
+fn import_account_corrupted_base64() {
```
[the remark this answers is not in the document]
Comment (Parth) [reply core/tests/account_service_tests.rs:103]: coverage?

File: core/tests/drawing_tests.rs [hunk 6]
```diff
@@ -23,6 +24,25 @@ fn parse_drawing_valid() {
...
+#[test]
+fn parse_drawing_valid() {
+    let drawing = Drawing {
+        scale: 0.0,
+        translation_x: 0.0,
+        translation_y: 0.0,
+        strokes: vec![Stroke {
+            points_x: vec![10f32, 50f32, 60f32],
+            points_y: vec![10f32, 50f32, 1000f32],
+            points_girth: vec![5f32, 7f32, 9f32],
```
Comment (steverusso) [inline core/tests/drawing_tests.rs:39]: nit: I see we had this before, but unless the `f32` suffix is necessary here, `.0` is cleaner and easier to read imo and is more consistent.

File: core/tests/account_service_tests.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (tvanderstad) [reply core/tests/account_service_tests.rs:103]: yes

File: core/tests/account_service_tests.rs (hunk 1 above)
Comment (tvanderstad) [reply core/tests/account_service_tests.rs:112]: confirms that this is valid base64, and therefore that the failure happens after base64 decoding

File: core/src/model/drawing.rs (hunk 3 above)
Comment (tvanderstad) [reply core/src/model/drawing.rs:27]: done!

File: core/src/model/drawing.rs (hunk 4 above)
Comment (tvanderstad) [reply core/src/model/drawing.rs:115]: done!

File: core/tests/account_service_tests.rs (hunk 2 above)
Comment (tvanderstad) [reply core/tests/account_service_tests.rs:164]: done!

File: core/tests/drawing_tests.rs (hunk 6 above)
Comment (tvanderstad) [reply core/tests/drawing_tests.rs:39]: done!

**Pull request merged**